### PR TITLE
MAISTRA-1292: update.sh: Exit upon errors

### PR DESCRIPTION
**Pull request by jwendell**
And make them more robust by adding some shell guards (-e, -u -o pipefail).

We should make sure it exits when encountering an error.

### CODE REVIEW

**Review by rcernich — APPROVED** [review]

File: bazel/update.sh [hunk 1]
```diff
@@ -35,7 +39,7 @@ function update_version() {
         echo "Already on disk, download not necessary"
     fi
 
-    sed -i "s/Version: .*/Version:         ${version}/" bazel.spec.spec
+    sed -i "s/Version: .*/Version:        ${version}/" bazel.spec
```
Comment (dgn) [inline bazel/update.sh:42]: nit: I guess we could match the spaces in a `(group)` and insert them in the replace part using `\1`, then we don't need to make sure the number of whitespaces is right (yes I've been doing too much `sed` lately)

File: istio-proxy/update.sh [hunk 2]
```diff
@@ -13,8 +17,7 @@ while getopts ":p:" opt; do
     esac
 done
 
-[[ -z "${PROXY_SHA}" ]] && PROXY_SHA="$(grep '%global proxy_git_commit ' istio-proxy.spec | cut -d' ' -f3)"
-
+PROXY_SHA=${PROXY_SHA-"$(grep '%global proxy_git_commit ' istio-proxy.spec | cut -d' ' -f3)"}
```
Comment (dgn) [inline istio-proxy/update.sh:20]: nit: why not also use `awk` here?

File: istio/update.sh [hunk 3]
```diff
@@ -15,7 +19,7 @@ while getopts ":i:v:" opt; do
   esac
 done
 
-[[ -z "${ISTIO_SHA}" ]] && ISTIO_SHA="$(grep '%global git_commit ' istio.spec | cut -d' ' -f3)"
+ISTIO_SHA=${ISTIO_SHA:-"$(grep '%global git_commit ' istio.spec | cut -d' ' -f3)"}
```
Comment (dgn) [inline istio/update.sh:22]: same as above

File: prometheus/update.sh [hunk 4]
```diff
@@ -15,7 +19,7 @@ while getopts ":i:" opt; do
   esac
 done
 
-[[ -z "${PROMETHEUS_SHA}" ]] && PROMETHEUS_SHA="$(grep '%global git_commit ' prometheus.spec | cut -d' ' -f3)"
+PROMETHEUS_SHA=${PROMETHEUS_SHA:-"$(grep '%global git_commit ' prometheus.spec | cut -d' ' -f3)"}
```
Comment (dgn) [inline prometheus/update.sh:22]: and here

**Review by dgn — APPROVED** [review]
lgtm, just minor comments

File: bazel/update.sh (hunk 1 above)
Comment (jwendell) [reply bazel/update.sh:42]: that's not the focus of this pr. I only changed strictly what was necessary for update.sh to succeed with my changes. Further improvements can be done in another PR.

File: istio-proxy/update.sh (hunk 2 above)
Comment (jwendell) [reply istio-proxy/update.sh:20]: that's not the focus of this pr. I only changed strictly what was necessary for update.sh to succeed with my changes. Further improvements can be done in another PR.

File: istio/update.sh (hunk 3 above)
Comment (jwendell) [reply istio/update.sh:22]: that's not the focus of this pr. I only changed strictly what was necessary for update.sh to succeed with my changes. Further improvements can be done in another PR.

File: prometheus/update.sh (hunk 4 above)
Comment (jwendell) [reply prometheus/update.sh:22]: that's not the focus of this pr. I only changed strictly what was necessary for update.sh to succeed with my changes. Further improvements can be done in another PR.

File: istio-proxy/update.sh (hunk 2 above)
Comment (dgn) [reply istio-proxy/update.sh:20]: but you did change it in the top of the file from `cut` to `awk`?

File: bazel/update.sh [hunk 5]
```diff
@@ -15,7 +19,7 @@ while getopts ":i:v:" opt; do
   esac
 done
 
-[[ -z "${BAZEL_VERSION}" ]] && BAZEL_VERSION="$(grep 'Version: ' bazel.spec | cut -d' ' -f9)"
+BAZEL_VERSION=${BAZEL_VERSION:-"$(grep 'Version: ' bazel.spec | awk '{print $2}')"}
```
Comment (dgn) [inline bazel/update.sh:22]: here

File: ior/update.sh [hunk 6]
```diff
@@ -15,7 +19,7 @@ while getopts ":i:v:" opt; do
   esac
 done
 
-[[ -z "${IOR_SHA}" ]] && IOR_SHA="$(grep '%global git_commit ' ior.spec | cut -d' ' -f3)"
```
Comment (brian-avery) [inline ior/update.sh:18]: Can this use awk as well?

File: istio-cni/update.sh [hunk 7]
```diff
@@ -15,7 +19,7 @@ while getopts ":i:v:" opt; do
   esac
 done
 
-[[ -z "${ISTIO_CNI_SHA}" ]] && ISTIO_CNI_SHA="$(grep '%global git_commit ' istio-cni.spec | cut -d' ' -f3)"
```
Comment (brian-avery) [inline istio-cni/update.sh:18]: same here

File: bazel/update.sh (hunk 5 above)
Comment (jwendell) [reply bazel/update.sh:22]: Because the cut was wrong.

File: istio-proxy/update.sh (hunk 2 above)
Comment (brian-avery) [reply istio-proxy/update.sh:20]: I agree with Daniel. Cut is fragile. Not sure it's strictly necessary for the first case, but regardless it cleans everything up. Lets do all or nothing.

**Pull request merged**